README.md

```diff
@@ -1,2 +1,105 @@
-# procedure-module-pattern
-In a system that processes Messages, place each Procedure in a separate module.
+# Procedure Module pattern
+
+**"A module containing all Phases of a message processing Procedure."**
+
+The **Procedure Module** is the short name for the **Message Processing Procedure Module (MPPM)**.
+
+The **Procedure Module** is a structural code organization pattern.
+
+## Definitions
+
+**Message**: information sent to notifies that something has occurred, or to request for something to occur. Actions, Commands and Events are Messages.  
+Messages are identified by their **type**, and the information they carry is called the message **payload**.  
+A Message may express the intent to do something, but must never express how to do it.
+
+**Procedure**: steps performed in a certain order to obtain a result.
+In the context of processing messages, when a Message of a certain type is received, a certain Procedure will process that Message.  
```

Comment:
Maybe clarify that the Procedure explains the "how", which is specifically excluded from the Message.

Reply:
done:
Steps performed in a certain order to obtain a result. A Procedure expresses how to do something.
In the context of processing messages, ...